src/components/select-user.tsx

```diff
@@ -0,0 +1,134 @@
+import { Fragment, useState } from "react";
+import Image from "next/image";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Listbox } from "@headlessui/react";
+import { useQuery } from "@tanstack/react-query";
+
+import HoverOpacityButton from "@/components/Button/hover-opacity-button";
+import Select from "@/components/select";
+import LoadingSpinner from "@/components/UI/loading-spinner";
+import { useUsers } from "@/hooks/users";
+import { httpClient } from "@/lib/base-http-client";
+import cn from "@/lib/classnames";
+import isBlank from "@/lib/is-blank";
+import { User } from "@/types";
+const NAME_KEY = "emailAddress";
+const VALUE_KEY = "id";
+interface SelectUserProps {
+  value?: User[];
+  onChange?: (value: User[]) => void;
+  multiple?: boolean;
+  label?: string;
+  initialUserIds?: string[];
+}
+
+// todo: improve UX
+// todo: add dynamic loading when scroll to bottom
+export default function SelectUser(props: SelectUserProps) {
+  const { initialUserIds, value, onChange, ...rest } = props;
+  const [search, setSearch] = useState<string>("");
+  const { data, isLoading } = useUsers(10, 1, "-created_at", search);
+  const users = data?.items as User[];
+  const [initialized, setInitialized] = useState<boolean>(false);
+
+  useQuery({
```

Comment:
Consider moving this out into `hooks/users.ts`, and pass all the stuff you need and a callback in the params
e.g.
```ts
useUserList(userIds, () => setInitialized(true))
```

Reply:
Actually, to avoid this initialized and onChange stuff, I'd instead fetch the users from the parent component and pass the desired list of users to render as a prop. In general, it's cleaner and more maintainable to keep ui components "dumb"